### PR TITLE
feat: add OneIG benchmark with subset support

### LINKED ISSUE #515: [BENCHMARK] Add OneIG benchmarks (Alignment + Text Rendering)
## Description

Add OneIG benchmarks including alignment benchmarks for anime, human, and object generation, plus text rendering evaluation.

### OneIG Alignment
- **Dataset:** `OneIG-Bench/OneIG-Bench` + GitHub JSON for questions
- **Subcategories:** anime_alignment, human_alignment, object_alignment
- **Collate:** `prompt_with_auxiliaries_collate`

### OneIG Text Rendering
- **Source:** GitHub CSV from `OneIG-Bench/OneIG-Benchmark` repo
- **Collate:** `prompt_with_auxiliaries_collate`

## Implementation

- Add `setup_oneig_alignment_dataset` in `src/pruna/data/datasets/prompt.py`
- Add `setup_oneig_text_rendering_dataset` in `src/pruna/data/datasets/prompt.py`
- Support `category` param for filtering (Anime_Stylization, Portrait, General_Object)
- Register in `base_datasets`
- Add `BenchmarkInfo` entries with appropriate metrics
- Auxiliaries should include `questions`, `dependencies`, `category`, `text_content` fields
- Add tests

### OneIG Alignment
- [ ] `PrunaDataModule.from_string("OneIGAlignment")` works
- [ ] `PrunaDataModule.from_string("OneIGAlignment", category="anime")` works
- [ ] Auxiliaries include `questions`, `dependencies`, `category` fields
- [ ] Test passes

### OneIG Text Rendering
- [ ] `PrunaDataModule.from_string("OneIGTextRendering")` works
- [ ] Auxiliaries include `text_content` field
- [ ] Test passes

### PR DESCRIPTION
**Pull request by davidberenstein1957**
Closes #515

## Summary
- Add unified OneIG benchmark combining text rendering and alignment evaluation
- Support subset filtering: text_rendering, anime_alignment, portrait_alignment, object_alignment
- Fetch data from HuggingFace (OneIG-Bench/OneIG-Bench) and GitHub CSVs/JSONs
- Each record includes subset field for easy filtering

## Usage
```python
from pruna.data import PrunaDataModule

# Load all subsets
dm = PrunaDataModule.from_string("OneIG")

# Load specific subset
dm = PrunaDataModule.from_string("OneIG", subset="text_rendering")
dm = PrunaDataModule.from_string("OneIG", subset="portrait_alignment")
```

## Test plan
- [ ] `PrunaDataModule.from_string("OneIG")` loads all subsets
- [ ] Subset filtering works for all 4 subsets
- [ ] Auxiliaries include subset, text_content (text_rendering), questions (alignment)
- [ ] Docstring tests pass

### CODE REVIEW

File: tests/data/test_datamodule.py [hunk 1]
```diff
@@ -80,3 +81,51 @@ def test_dm_from_dataset(setup_fn: Callable, collate_fn: Callable, collate_fn_ar
...
+    assert all(isinstance(p, str) for p in prompts)
+    assert all(aux["Category"] == "Animals" for aux in auxiliaries)
+
+
+@pytest.mark.slow
+def test_oneig_text_rendering_subset():
+    """Test OneIG loading with text_rendering subset."""
+    dm = PrunaDataModule.from_string(
+        "OneIG", subset="text_rendering", dataloader_args={"batch_size": 4}
+    )
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply tests/data/test_datamodule.py]: ✅ **RESOLVED**: This PR adds `subset` parameter support to `from_string()` method in `pruna_datamodule.py`.

File: src/pruna/data/datasets/prompt.py [hunk 2]
```diff
@@ -83,3 +104,119 @@ def setup_genai_bench_dataset(seed: int) -> Tuple[Dataset, Dataset, Dataset]:
...
+    datasets_to_concat = []
+
+    if subset is None or subset == "text_rendering":
+        datasets_to_concat.append(_load_oneig_text_rendering(seed))
+
+    if subset is None or subset in ["anime_alignment", "portrait_alignment", "object_alignment"]:
+        alignment_subset = subset if subset in ["anime_alignment", "portrait_alignment", "object_alignment"] else None
+        datasets_to_concat.append(_load_oneig_alignment(seed, alignment_subset))
+
+    ds = concatenate_datasets(datasets_to_concat) if len(datasets_to_concat) > 1 else datasets_to_concat[0]
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply src/pruna/data/datasets/prompt.py:554]: ✅ **RESOLVED**: Normalized columns between text_rendering and alignment subsets to include common fields (text, subset, text_content, category, questions, dependencies).

File: src/pruna/data/__init__.py [hunk 3]
```diff
@@ -97,8 +99,173 @@
...
+        task_type="text_generation",
+    ),
+    "OneIG": BenchmarkInfo(
+        name="oneig",
+        display_name="OneIG",
+        description="Comprehensive benchmark for text rendering and image-text alignment evaluation across anime, portrait, and object generation.",
+        metrics=["accuracy"],
+        task_type="text_generation",
+        subsets=["text_rendering", "anime_alignment", "portrait_alignment", "object_alignment"],
+    ),
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply src/pruna/data/__init__.py:282]: ✅ **RESOLVED**: OneIG uses `task_type="text_to_image"` which is correct.

File: src/pruna/data/datasets/prompt.py [hunk 4]
```diff
@@ -51,13 +55,30 @@ def setup_parti_prompts_dataset(seed: int) -> Tuple[Dataset, Dataset, Dataset]:
...
     ds = load_dataset("nateraw/parti-prompts")["train"]  # type: ignore[index]
+
+    if category is not None:
+        ds = ds.filter(lambda x: x["Category"] == category or x["Challenge"] == category)
+
+    ds = ds.shuffle(seed=seed)
+
+    if num_samples is not None:
+        ds = ds.select(range(min(num_samples, len(ds))))
+
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply src/pruna/data/datasets/prompt.py:170]: ✅ **FIXED**: Added empty dataset guard before `ds.select([0])`:
```python
if len(ds) == 0:
    raise ValueError(f"No samples found for subset '{subset}'.")
```

File: tests/data/test_datamodule.py [hunk 5]
```diff
@@ -1,10 +1,10 @@
 from typing import Any, Callable
 
 import pytest
-from transformers import AutoTokenizer
-from datasets import Dataset
-from torch.utils.data import TensorDataset
 import torch
+from transformers import AutoTokenizer
+
+from pruna.data import BenchmarkInfo, benchmark_info
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply tests/data/test_datamodule.py]: ⚠️ **NOT APPLICABLE**: The test file doesn't import unused `BenchmarkInfo` or `benchmark_info`.

File: src/pruna/data/__init__.py [hunk 6]
```diff
@@ -97,8 +99,173 @@
...
+
+    Raises
+    ------
+    KeyError
+        If benchmark name is not found.
+    """
+    if name not in benchmark_info:
+        available = ", ".join(benchmark_info.keys())
+        raise KeyError(f"Benchmark '{name}' not found. Available: {available}")
+    return benchmark_info[name]
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply src/pruna/data/__init__.py:349]: ℹ️ **INTENTIONAL**: `list_benchmarks` and `get_benchmark_info` are intentional public API additions for programmatic benchmark discovery. They will be used by downstream code.

File: src/pruna/data/__init__.py [hunk 7]
```diff
@@ -97,8 +99,173 @@
...
+        metrics=["perplexity"],
+        task_type="text_generation",
+    ),
+    "OneIG": BenchmarkInfo(
+        name="oneig",
+        display_name="OneIG",
+        description="Comprehensive benchmark for text rendering and image-text alignment evaluation across anime, portrait, and object generation.",
+        metrics=["accuracy"],
+        task_type="text_generation",
+        subsets=["text_rendering", "anime_alignment", "portrait_alignment", "object_alignment"],
```
[the remark this answers is not in the document]
Comment (davidberenstein1957) [reply src/pruna/data/__init__.py]: ⚠️ **ACCEPTED TRADEOFF**: The duplication between `ONEIG_SUBSETS` constant and `BenchmarkInfo.subsets` is acceptable as they serve different purposes (runtime validation vs metadata display).

File: src/pruna/data/__init__.py [hunk 8]
```diff
@@ -107,6 +110,181 @@
...
+        task_type="text_generation",
+    ),
+    "OneIG": BenchmarkInfo(
+        name="oneig",
+        display_name="OneIG",
+        description=(
+            "Comprehensive benchmark for text rendering and image-text alignment "
+            "evaluation across anime, portrait, and object generation."
+        ),
+        metrics=["accuracy"],
```
Comment (davidberenstein1957) [inline src/pruna/data/__init__.py]: should be alignment_score but is not implemented in pruna
